### PR TITLE
update nixpkgs

### PR DESCRIPTION
**Pull request by nixbitcoin**
Now that unstable contains a bunch of pkg updates I thought now might also be a good time to update stable.

### CODE REVIEW

**Comment by erikarvstedt** [comment]
[Here are some patches](https://github.com/erikarvstedt/nix-bitcoin/commits/update-nixpkgs-ea) that are partially dependent on this PR. Could you add them here?

**Comment by nixbitcoin** [comment]
Done. Also had to patch spark-wallet because it was failing to build after I cleared out my `/nix/store`.

**Review by erikarvstedt — APPROVED** [review]
ACK b3e868d0af9a1b724d8466affbaa4c7d5454084d

File: modules/bitcoind-rpc-public-whitelist.nix [hunk 1]
```diff
@@ -25,7 +25,6 @@
   "gettxoutsetinfo"
   "scantxoutset"
   "verifytxoutproof"
-  "waitfornewblock"
```
Comment (jonasnick) [inline modules/bitcoind-rpc-public-whitelist.nix:28]: I don't think that the whitelist is minimal for our services right now. If we wanted to do that we could remove quite a bit more. It consists of RPCs that are rather benign and `waitfornewblock` seems to fall in that category. So I'd rather keep it in. If we wanted to improve on this we could have individual whitelists per service.

Comment (erikarvstedt) [reply modules/bitcoind-rpc-public-whitelist.nix:28]: The main reason for removing it is that the old whitelist inspired https://github.com/romanz/electrs/issues/522 and the subsequent fix in electrs.
I'll let you decide.

Comment (erikarvstedt) [reply modules/bitcoind-rpc-public-whitelist.nix:28]: `waitfornewblock` is [not recommnended for public use](https://github.com/bitcoin/bitcoin/pull/10519#issuecomment-323094194). The old whitelist inspired https://github.com/romanz/electrs/issues/522 and the subsequent fix in electrs. I'll let you decide.

Comment (jonasnick) [reply modules/bitcoind-rpc-public-whitelist.nix:28]: There's a difference between public for nix-bitcoin services and public for the rest of the world. But I see that it's generally recommended to not use it. So it's fine by me to remove it (also, we're able to notice lower quality applications trying to use waitfornewblock).

**Review by jonasnick — COMMENTED** [review]
utACK b3e868d0af9a1b724d8466affbaa4c7d5454084d

**Pull request merged**